docs/_docs/reference/changed-features/pattern-matching.md

```diff
@@ -162,27 +142,66 @@ object Nat:
 // 5 is a natural number
 ```
 
-## Name-based Match
+### Name-based Match
 
-- `N > 1` is the maximum number of consecutive (parameterless `def` or `val`) `_1: P1 ... _N: PN` members in `U`
+- `S` has `N > 1` members such that they are each `val`s or parameterless `def`s, and named from `_1` with type `P1` to `_N` with type `PN`
+- `S` doesn't have `N+1` members satisfying the previous point, `N` is maximal
 - Pattern-matching on exactly `N` patterns with types `P1, P2, ..., PN`
 
+For example, where `AlwaysEmpty <: R`, `S` is `NameBased`:
 ```scala
-object ProdEmpty:
+object MyPatternMatcher:
+  def unapply(s: String) = AlwaysEmpty
+
+object AlwaysEmpty:
+  def isEmpty = true
+  def get = NameBased
```

Comment:
Consider adding explicit result types here. When reading code without an IDE, it is particularly important.

Reply:
```suggestion
object MyPatternMatcher:
  def unapply(s: String): AlwaysEmpty.type = AlwaysEmpty

object AlwaysEmpty:
  def isEmpty: Boolean = true
  def get: NameBased.type  = NameBased
```
In this case, I feel like it doesn't help, since the RHS is already a singleton

Reply:
@sjrd Seeing the above, is the current way acceptable ?